src/products/dto/create-product.dto.ts

```diff
@@ -12,18 +12,22 @@ export class CreateProductDto {
     @IsEnum(['MALE','FEMALE', 'NON_BINARY'])
     readonly productGender: [];
   
+    @ValidateIf((object, value) => object.productType === 'Shoes')
     @IsOptional()
```

Comment:
Remove @IsOptional decorator

line 15 validates if object.productType matches with 'Shoes'. If it does match, then the next decorator is checked. @IsOptional() will create an entry in the database with an empty array if object.productSizeShoe is empty or is not present in the object payload. Removing @IsOptional decorator will allow @isEnum decorator to run so value of productSizeShoe's value is one of the enum values.

Example
- The image below shows a post request with no productShoeSize in payload. 
![CleanShot 2024-07-08 at 15 50 01](https://github.com/SeattleColleges/belindas-closet-nestjs/assets/36749051/c9b3faea-6551-4f51-a454-bb88cbe89de4)
- Removing @IsOptional decorator will return the error below. Doing so will allow the @isEnum decorator to run and validate the payload.
![CleanShot 2024-07-08 at 15 56 07](https://github.com/SeattleColleges/belindas-closet-nestjs/assets/36749051/509265f3-87b2-4daf-8892-f987bd95b987)